File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int n  =s.length();
        stringstream ss(s);
        string token = "";
        string result ="";
        while(ss>>token){
            result = token + " " +result;
        }
        int m = result.length();
        return result.substr(0,m-1);
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] arr = s.trim().split(" ");
        int n =arr.length;
        StringBuilder sb = new StringBuilder();
        for(int i=n-1;i>=0;i-- ){
            if (arr[i].isEmpty())
                continue;
            sb.append(" " + arr[i]);
        }
        return sb.substring(1).toString();
    }
}